misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/AbstractCharSequenceToContainAtLeastExpectationsTest.kt

```diff
@@ -0,0 +1,362 @@
+        it("${toContainAtLeastButAtMostPair.first("'o'", "once", "twice")} does not throw") {
+            expect(helloWorld).toContainAtLeastButAtMostFun(1, 2, 'o')
+        }
+        it(
+            "${toContainAtLeastButAtMostPair.first("'o' and 'l'", "once", "twice")} throws AssertionError " +
+                "and message contains both, at most: 2 and how many times it actually contained 'l' (3)"
+        ) {
+            expect {
+                expect(helloWorld).toContainAtLeastButAtMostFun(1, 2, 'o', 'l')
+            }.toThrow<AssertionError> {
+                message {
+                    toContain(
+                        "$rootBulletPoint$toContainDescr: $separator" +
+                            "$valueWithIndent: 'l'",
+                        "$numberOfOccurrences: 3$separator"
+                    )
+                    toEndWith("$atMost: 2")
+                    notToContain(atLeast, "$valueWithIndent: 'o'")
+                }
+            }
+        }
+        it("${toContainAtLeastButAtMostPair.first("'o' and 'l'", "twice", "3 times")} does not throw") {
+            expect(helloWorld).toContainAtLeastButAtMostFun(2, 3, 'o', 'l')
+        }
+        it(
+            "${toContainAtLeastButAtMostPair.first("'o' and 'l'", "3 times", "4 times")} throws " +
+                "AssertionError and message contains both, at least: 3 and how many times it actually contained 'o' (2)"
+        ) {
+            expect {
+                expect(helloWorld).toContainAtLeastButAtMostFun(3, 4, 'o', 'l')
+            }.toThrow<AssertionError> {
+                message {
+                    toContain(
+                        "$rootBulletPoint$toContainDescr: $separator" +
+                            "$valueWithIndent: 'o'",
+                        "$numberOfOccurrences: 2$separator"
+                    )
+                    toEndWith("$atLeast: 3")
+                    notToContain(atMost, "$valueWithIndent: 'l'")
+                }
+            }
+        }
+    }
+
+    @TestFactory
+    fun toContainAtLeastButAtMostIgnoringCase() =
+        testFactory(toContainAtLeastButAtMostIgnoringCaseSpec) { toContainAtLeastButAtMostIgnoringCaseFun ->
+            it(
+                "${toContainAtLeastButAtMostIgnoringCasePair.first("'o' and 'l'", "twice", "3 times")} " +
+                    "does not throw"
+            ) {
+                expect(helloWorld)
+                    .toContainAtLeastButAtMostIgnoringCaseFun(2, 3, 'o', 'l')
+            }
+
+
+            it("${toContainAtLeastIgnoringCasePair.first("'o' and 'l'", " 3 times")} does not throw") {
+                expect(helloWorld)
+                    .toContainAtLeastButAtMostIgnoringCaseFun(3, 4, 'o', 'l')
+            }
+        }
+
+    companion object {
+        val text: CharSequence = "Hello my name is Robert"
+        val helloWorld: CharSequence = "Hello World, I am Oskar"
```

Comment:
something I overlooked in the past PRs we don't need to repeat them here, you can import `text` and `helloWorld` from CharSequenceToContainSpecBase